src/battle/actor/goomba.c

```diff
@@ -0,0 +1,488 @@
+
+// 432198-432474 (VRAM: 802197F8)
+Script goomba_idle = SCRIPT({
+10:
+    RandInt(80, SI_VAR(0));
+    SI_VAR(0) += 80;
+    loop SI_VAR(0) {
+0:
+        GetStatusFlags(ActorID_SELF, SI_VAR(1));
+        if (SI_VAR(1) & 0x35D000) {
+            sleep 1;
+            goto 0;
+        }
+        sleep 1;
+    }
+    GetActorPos(ActorID_SELF, SI_VAR(0), SI_VAR(1), SI_VAR(2));
+    SI_VAR(0) += 5;
+    SetActorIdleSpeed(ActorID_SELF, 1.0);
+    SetIdleAnimations(ActorID_SELF, 1, goomba_anims_running);
+    SetIdleGoal(ActorID_SELF, SI_VAR(0), SI_VAR(1), SI_VAR(2));
+    IdleRunToGoal(ActorID_SELF, 0);
+    SetIdleAnimations(ActorID_SELF, 1, goomba_anims);
+    loop 20 {
+1:
+        GetStatusFlags(ActorID_SELF, SI_VAR(1));
+        if (SI_VAR(1) & 0x35D000) {
+            sleep 1;
+            goto 1;
+        }
+        sleep 1;
+    }
+    GetActorPos(ActorID_SELF, SI_VAR(0), SI_VAR(1), SI_VAR(2));
+    SI_VAR(0) -= 5;
+    SetActorIdleSpeed(ActorID_SELF, 1.0);
+    SetIdleAnimations(ActorID_SELF, 1, goomba_anims_running);
+    SetIdleGoal(ActorID_SELF, SI_VAR(0), SI_VAR(1), SI_VAR(2));
+    IdleRunToGoal(ActorID_SELF, 0);
+    SetIdleAnimations(ActorID_SELF, 1, goomba_anims);
+    loop 80 {
+2:
+        GetStatusFlags(ActorID_SELF, SI_VAR(1));
+        if (SI_VAR(1) & 0x35D000) {
+            sleep 1;
+            goto 2;
+        }
+        sleep 1;
+    }
+    goto 10;
+});
+
+// 432474-432B34 (VRAM: 80219AD4)
+Script goomba_dispatch = SCRIPT({
+    UseIdleAnimation(ActorID_SELF, 0);
+    EnableIdleScript(ActorID_SELF, 0);
+    SetActorScale(ActorID_SELF, 1.0, 1.0, 1.0);
+    GetLastEvent(ActorID_SELF, SI_VAR(0));
+    match SI_VAR(0) {
+        Event_HIT_COMBO, Event_HIT {
+            SI_VAR(0) = c 0x1;
+            SI_VAR(1) = c NPC_ANIM(goomba, normal, pain);
+            await DoNormalHit;
+        }
+        == Event_BURN_HIT {
+            SI_VAR(0) = c 0x1;
+            SI_VAR(1) = c NPC_ANIM(goomba, normal, burn_pain);
+            SI_VAR(2) = c NPC_ANIM(goomba, normal, burn_dead);
+            await DoBurnHit;
+        }
+        == Event_BURN_DEATH {
+            SI_VAR(0) = c 0x1;
+            SI_VAR(1) = c NPC_ANIM(goomba, normal, burn_pain);
+            SI_VAR(2) = c NPC_ANIM(goomba, normal, burn_dead);
+            await DoBurnHit;
+            SI_VAR(0) = c 0x1;
+            SI_VAR(1) = c NPC_ANIM(goomba, normal, burn_dead);
+            await DoDeath;
+            return;
+        }
+        == Event_SPIN_SMASH_HIT {
+            SI_VAR(0) = c 0x1;
+            SI_VAR(1) = c NPC_ANIM(goomba, normal, pain);
+            await DoSpinSmashHit;
+        }
+        == Event_SPIN_SMASH_DEATH {
+            SI_VAR(0) = c 0x1;
+            SI_VAR(1) = c NPC_ANIM(goomba, normal, pain);
+            await DoSpinSmashHit;
+            SI_VAR(0) = c 0x1;
+            SI_VAR(1) = c NPC_ANIM(goomba, normal, dead);
+            await DoDeath;
+            return;
+        }
+        == Event_SHOCK_HIT {
+            SI_VAR(0) = c 0x1;
```

Comment:
probably just 1 instead of 0x1 (also relevant elsewhere)

Reply:
this may actually be a bool, I'm not sure what the DoX scripts expect